Compile in release mode must be silent

When the game is compiled in release, there are multiple compiler warnings, such as [these](https://travis-ci.org/github/tresinformal/game/builds/690103731#L1520):

```
ame.cpp: In function ‘void test_game()’:
game.cpp:388:22: warning: unused variable ‘before’ [-Wunused-variable]
game.cpp:390:22: warning: unused variable ‘after’ [-Wunused-variable]
game.cpp:399:22: warning: unused variable ‘before’ [-Wunused-variable]
[...]
```

These 'unused' variables are unused, because in release mode, all `assert`s are removed. Therefore, all what is left, are some variables.

The way to fix this, is to remove the tests, using an `#ifdef` on `NDEBUG`, such as:

```
#ifndef NDEBUG
//  tests here
#endif
```

Put all tests in these `#ifndef`s. If there are unused variables elsewhere, ignore these for now.

--- a/action_type.cpp
+++ b/action_type.cpp
@@ -4,6 +4,7 @@
 
 void test_action_type()
 {
+  #ifndef NDEBUG // no tests is release
   // to_str
   {
     assert(to_str(action_type::turn_left) == "turn_left");
@@ -21,6 +22,7 @@ void test_action_type()
     s << action_type::brake;
     assert(!s.str().empty());
   }
+#endif
 }
 
 std::ostream &operator<<(std::ostream &os, const action_type t)

--- a/color.cpp
+++ b/color.cpp
@@ -43,6 +43,7 @@ std::string to_str(const color& in_color) noexcept
 
 void test_color()
 {
+    #ifndef NDEBUG // no tests is release
     // Color is white by default
     {
         const color c;
@@ -82,6 +83,7 @@ void test_color()
         const color c;
         std::cout << c;
     }
+#endif
 }
 
 bool operator==(const color& lhs, const color& rhs) noexcept

--- a/enemy.cpp
+++ b/enemy.cpp
@@ -4,6 +4,7 @@
 
 void test_enemy()
 {
+  #ifndef NDEBUG // no tests is release
   {
     const enemy e;
     assert(e.get_x() == 0.0);
@@ -29,6 +30,7 @@ void test_enemy()
     const enemy e;
     s << e;
   }
+#endif
 }
 
 enemy::enemy(const double x, const double y) : m_x{x}, m_y{y} {}

--- a/enemy_behavior_type.cpp
+++ b/enemy_behavior_type.cpp
@@ -7,6 +7,7 @@
 
 void test_enemy_behavior_type()
 {
+  #ifndef NDEBUG // no tests is release
   const enemy_behavior_type t{enemy_behavior_type::gezellig};
   const enemy_behavior_type u{enemy_behavior_type::shy};
   static_assert(t != u,"");
@@ -29,7 +30,7 @@ void test_enemy_behavior_type()
     std::stringstream s;
     assert(s.str().empty());
   }
-
+  #endif
 }
 
 std::ostream &operator<< (std::ostream &out, const enemy_behavior_type &enemy_type)

--- a/environment.cpp
+++ b/environment.cpp
@@ -30,6 +30,7 @@ std::ostream &operator<<(std::ostream &os, const environment& e)
 
 void test_environment()
 {
+  #ifndef NDEBUG // no tests is release
   // Minimal and maximal coordinats should make sense
   {
     const environment e;
@@ -62,4 +63,5 @@ void test_environment()
     assert(!s.str().empty());
   }
   #endif // FIX_ISSUE_204
+#endif // no tests in release
 }

--- a/environment_type.cpp
+++ b/environment_type.cpp
@@ -25,6 +25,7 @@ std::string to_str(environment_type this_env_type)
 
 void test_individual_type()
 {
+  #ifndef NDEBUG // no tests is release
   // Conversion to string
   {
     assert(to_str(environment_type::empty) == "empty");
@@ -39,6 +40,7 @@ void test_individual_type()
     std::stringstream s;
     s << environment_type::empty;
  }
+#endif
 }
 
 std::ostream &operator<<(std::ostream &os, const environment_type t)

--- a/food.cpp
+++ b/food.cpp
@@ -20,7 +20,7 @@ std::ostream &operator<<(std::ostream &os, const food f)
 
 void test_food()
 {
-
+  #ifndef NDEBUG // no tests in release
   {
     const food f;
     assert(f.get_x() == 0.0);
@@ -86,9 +86,7 @@ void test_food()
     assert(f.get_regeneration_time() == regeneration_time);
   }
 #endif
-
-
-
+#endif // no tests in release
 }
 
 

--- a/food_state.cpp
+++ b/food_state.cpp
@@ -5,7 +5,9 @@
 
 void test_food_state()
 {
+ #ifndef NDEBUG
  {
    assert(food_state::eaten!= food_state::uneaten);
  }
+ #endif
 }

--- a/game.cpp
+++ b/game.cpp
@@ -442,6 +442,7 @@ player game::wall_collision(player p)
 
 void test_game() //!OCLINT tests may be many
 {
+  #ifndef NDEBUG // no tests in release
   // The game has done zero ticks upon startup
   {
     const game g;
@@ -1081,6 +1082,6 @@ void test_game() //!OCLINT tests may be many
     assert(!nth_food_is_eaten(g,0));
   }
 #endif
-
+#endif // no tests in release
 }
 

--- a/game_options.cpp
+++ b/game_options.cpp
@@ -25,7 +25,7 @@ void music_on(game_options& o) noexcept
 
 void test_game_options()
 {
-
+  #ifndef NDEBUG // no tests in release
   // By default, music is played
   {
     const game_options g;
@@ -65,4 +65,5 @@ void test_game_options()
     a.get_kam_1();
   }
   #endif // FIX_ISSUE_268
+  #endif
 }

--- a/game_resources.cpp
+++ b/game_resources.cpp
@@ -162,12 +162,12 @@ game_resources::game_resources()
 
 void test_game_resources()
 {
+  #ifndef NDEBUG // no tests in release
   game_resources g;
   assert(g.get_heterogenous_landscape().getSize().x > 0.0);
 
   #ifdef FIX_ISSUE_136
   assert(g.get_sound(sound_type::shoot).getDuration().asMicroseconds() > 0.0);
   #endif
-
-
+  #endif
 }

--- a/game_view.cpp
+++ b/game_view.cpp
@@ -285,6 +285,7 @@ bool is_nth_player_stunned(const game_view& g, const int& p) noexcept
 
 void test_game_view()//!OCLINT tests may be many
 {
+    #ifndef NDEBUG // no tests in release
     {
         // Show the game for one frame
         // (there will be a member function 'exec' for running the game)
@@ -416,7 +417,7 @@ void test_game_view()//!OCLINT tests may be many
     g.process_events(); // Needed to process the event
     assert(is_nth_player_stunned(g, 0));
   }
-
+  #endif
 }
 
 

--- a/key_action_map.cpp
+++ b/key_action_map.cpp
@@ -58,6 +58,7 @@ bool key_action_map::has_key(sf::Keyboard::Key key) const noexcept
 
 void test_key_action_map()//!OCLINT tests can be many
 {
+    #ifndef NDEBUG // no tests in release
     //It is possible to return an action type from a key being pressed
     //The keys are for now the ones pressed by player1 (position 0 in the player index)
     {
@@ -123,5 +124,5 @@ void test_key_action_map()//!OCLINT tests can be many
         assert(action_type::none == m.to_action(sf::Keyboard::L));
 
     }
-
+    #endif
 }

--- a/main.cpp
+++ b/main.cpp
@@ -28,6 +28,7 @@
 /// All tests are called from here, only in debug mode
 void test()
 {
+#ifndef NDEBUG
   test_optional();
   test_action_type();
   test_player_shape();
@@ -56,6 +57,7 @@ void test()
   test_game_resources();
   test_sound_type();
 #endif // LOGIC_ONLY
+#endif
 }
 
 int main(int argc, char **argv) //!OCLINT tests may be long

--- a/menu.cpp
+++ b/menu.cpp
@@ -39,7 +39,7 @@ void menu::put_buttons_tidy() noexcept
 
 void test_menu()
 {
-
+  #ifndef NDEBUG // no tests in release
   // menu_view is built with values
   // for width and height of the screen
   {
@@ -73,4 +73,5 @@ void test_menu()
              (static_cast<int>(v.get_buttons().size()) + 1) <
              0.000001f);
   }
+  #endif
 }

--- a/menu_button.cpp
+++ b/menu_button.cpp
@@ -18,6 +18,7 @@ const sf::Vector2f &menu_button::get_body() const noexcept { return m_body; }
 
 void test_menu_button()
 {
+    #ifndef NDEBUG // no tests in release
     // buttons can be set to a certain position
     {
         float x = 1.2f;
@@ -39,4 +40,5 @@ void test_menu_button()
                 && m_b.get_body().y == height
                && m_b.get_name() == name);
     }
+#endif
 }

--- a/optional.cpp
+++ b/optional.cpp
@@ -6,6 +6,7 @@
 
 void test_optional()
 {
+  #ifndef NDEBUG // no tests in release
   // Initialize with a value
   {
     const int value = 42;
@@ -35,6 +36,7 @@ void test_optional()
       //Great! x should throw, as there is no value
     }
   }
+  #endif
 }
 
 

--- a/player.cpp
+++ b/player.cpp
@@ -196,6 +196,7 @@ void remove_action(player& p, action_type action) noexcept
 
 void test_player() //!OCLINT tests may be long
 {
+  #ifndef NDEBUG // no tests in release
   // Can default construct a player
   {
     const player p;
@@ -630,4 +631,5 @@ void test_player() //!OCLINT tests may be long
     assert(is_stunned(p));
   }
 #endif
+#endif // no tests in release
 }

--- a/player_shape.cpp
+++ b/player_shape.cpp
@@ -6,12 +6,14 @@
 
 void test_player_shape()
 {
+  #ifndef NDEBUG // no tests in release
   // Conversion to string
   {
     assert(to_str(player_shape::circle) == "circle");
     assert(to_str(player_shape::rocket) == "rocket");
     assert(to_str(player_shape::square) == "square");
   }
+  #endif
 }
 
 std::string to_str(player_shape this_player_shape)

--- a/player_state.cpp
+++ b/player_state.cpp
@@ -3,6 +3,7 @@
 
 void test_player_state()
 {
+  #ifndef NDEBUG // no tests in release
   assert(player_state::active != player_state::stunned);
   {
     assert(player_state::dead != player_state::stunned &&
@@ -16,4 +17,5 @@ void test_player_state()
     assert(to_str(player_state::stunned) == "stunned");
   }
   #endif // FIX_ISSUE_276
+  #endif
 }

--- a/program_state.cpp
+++ b/program_state.cpp
@@ -4,6 +4,7 @@
 
 void test_program_state()
 {
+  #ifndef NDEBUG // no tests in release
   assert(to_str(program_state::menu) !=
          to_str(program_state::paused));
   assert(to_str(program_state::paused) !=
@@ -14,6 +15,7 @@ void test_program_state()
     std::stringstream s;
     s << program_state::menu;
   }
+  #endif
 }
 
 std::ostream &operator<<(std::ostream &os, const program_state t)

--- a/projectile.cpp
+++ b/projectile.cpp
@@ -18,6 +18,7 @@ void projectile::move()
 
 void test_projectile()
 {
+  #ifndef NDEBUG // no tests in release
   // Constructor
   {
     const double x{1.2};
@@ -32,5 +33,5 @@ void test_projectile()
     assert(t == p.get_type());
     assert(r == p.get_radius());
   }
-
+  #endif
 }

--- a/projectile_type.cpp
+++ b/projectile_type.cpp
@@ -4,6 +4,7 @@
 
 void test_projectile_type()
 {
+  #ifndef NDEBUG // no tests in release
   // String conversion
   {
     assert(to_str(projectile_type::cat) == "cat");
@@ -15,6 +16,7 @@ void test_projectile_type()
     s << projectile_type::rocket;
     assert(!s.str().empty());
   }
+  #endif
 }
 
 std::ostream &operator<<(std::ostream &os, const projectile_type t)

--- a/shelter.cpp
+++ b/shelter.cpp
@@ -81,6 +81,7 @@ const std::string to_str(const shelter& in_shelter) noexcept
 }
 void test_shelter() //!OCLINT tests may be complex
 {
+  #ifndef NDEBUG // no tests in release
   {
     shelter f;
     assert(f.get_x() == 0.0);
@@ -125,4 +126,5 @@ void test_shelter() //!OCLINT tests may be complex
     assert(!t.empty());
   }
   #endif // FIX_ISSUE_264
+  #endif
 }

--- a/sound_type.cpp
+++ b/sound_type.cpp
@@ -3,6 +3,7 @@
 
 void test_sound_type()
 {
+  #ifndef NDEBUG // no tests in release
   static_assert(sound_type::shoot != sound_type::hit, "");
   //#define FIX_ISSUE_263
   #ifdef FIX_ISSUE_263
@@ -12,6 +13,6 @@ void test_sound_type()
     assert(to_str(sound_type::hit) == "hit");
   }
   #endif // FIX_ISSUE_263
-
+  #endif
 }
 

